DNN Platform/Modules/Journal/ServicesController.cs

```diff
@@ -1,619 +1,621 @@
+        public HttpResponseMessage GetSuggestions(string keyword)
+        {
+            try
+            {
+                var findedUsers = new List<SuggestDTO>();
+                var relations = RelationshipController.Instance.GetUserRelationships(this.UserInfo);
+                foreach (var ur in relations)
+                {
+                    var targetUserId = ur.UserId == this.UserInfo.UserID ? ur.RelatedUserId : ur.UserId;
+                    var targetUser = UserController.GetUserById(this.PortalSettings.PortalId, targetUserId);
+                    var relationship = RelationshipController.Instance.GetRelationship(ur.RelationshipId);
+                    if (ur.Status == RelationshipStatus.Accepted && targetUser != null
+                        && ((relationship.RelationshipTypeId == (int)DefaultRelationshipTypes.Followers && ur.RelatedUserId == this.UserInfo.UserID)
+                                || relationship.RelationshipTypeId == (int)DefaultRelationshipTypes.Friends)
+                        && (targetUser.DisplayName.ToLowerInvariant().Contains(keyword.ToLowerInvariant())
+                                || targetUser.DisplayName.ToLowerInvariant().Contains(keyword.Replace("-", " ").ToLowerInvariant()))
+                        && findedUsers.All(s => s.userId != targetUser.UserID))
+                    {
+                        findedUsers.Add(new SuggestDTO
+                        {
+                            displayName = targetUser.DisplayName.Replace(" ", "-"),
+                            userId = targetUser.UserID,
+                            avatar = targetUser.Profile.PhotoURL,
+                            key = keyword,
+                        });
+                    }
+                }
+
+                return this.Request.CreateResponse(HttpStatusCode.OK, findedUsers.Cast<object>().Take(5));
+            }
+            catch (Exception exc)
+            {
+                Logger.Error(exc);
+                return this.Request.CreateErrorResponse(HttpStatusCode.InternalServerError, exc);
+            }
+        }
+
+        private static bool IsImageFile(string relativePath)
+        {
+            if (relativePath == null)
+            {
+                return false;
+            }
+
+            if (relativePath.Contains("?"))
+            {
+                relativePath = relativePath.Substring(
+                    0,
+                    relativePath.IndexOf("?", StringComparison.InvariantCultureIgnoreCase));
+            }
+
+            var extension = relativePath.Substring(relativePath.LastIndexOf(
+                ".",
+                StringComparison.Ordinal) + 1).ToLowerInvariant();
+            return AcceptedFileExtensions.Contains(extension);
+        }
+
+        private static bool IsAllowedLink(string url)
+        {
+            Uri uriResult;
+            return Uri.TryCreate(url, UriKind.Absolute, out uriResult)
+                && uriResult.Scheme != Uri.UriSchemeFile;

```

Comment:
Actually re-thinking about this, it is best practice to use allow-lists instead of deny-lists from a security perspective...

```suggestion
            return Uri.TryCreate(url, UriKind.Absolute, out uriResult)
                && (
                    uriResult.Scheme == Uri.UriSchemeHttp ||
                    uriResult.Scheme == Uri.UriSchemeHttps ||
                    uriResult.Scheme == Uri.UriSchemeFtp ||
                    uriResult.Scheme == Uri.UriSchemeMailto ||
                    uriResult.Scheme == Uri.UriSchemeNews ||
                    uriResult.Scheme == Uri.UriSchemeNntp
                );
```

Reply:
This seems more explicit.  Thanks @valadas 🎉 🌮

Reply:
CC: @mitchelsellers @bdukes

Reply:
this is what I was thinking of building from scratch, gave up and made the simple fix I put here.  I was hoping some other community ideas would make this right.

Reply:
The other solution works and the uri namespace is what I need to get familiar with.  I have a few more tweaks I want to do using pretty much this logic.

Reply:
I tested the first solution ok.  Testing your new one now.

Reply:
Seems to work good, thank you!

Reply:
if everyone agrees and nothing better is on its way anytime soon I will commit suggestion.

Reply:
@thabaum we chatted and would like to have @mitchelsellers and/or @bdukes review before committing this suggestion.  Thanks.

Reply:
as far as allow list

> Change to use an allow-list instead of a deny-list

I believe this is pretty much what I did.

Reply:
Do we want to support all of those protocols? I would think `http`, `https`, and `mailto` would be all we'd need

Reply:
I don't really mind, just went for everything in that namespace that was not something local... I could see an FTP link to download something off a public FTP for instance...